src/test/java/com/datatorrent/netlet/AbstractClientTest.java

```diff
@@ -127,16 +127,27 @@ public void sendData()
   {
   }
 
-  @SuppressWarnings("SleepWhileInLoop")
-  private void verifySendReceive(final DefaultEventLoop el, final int port) throws IOException, InterruptedException
+  @SuppressWarnings( {"SleepWhileInLoop", "AssertEqualsBetweenInconvertibleTypes"})
+  private void verifySendReceive(final DefaultEventLoop el) throws IOException, InterruptedException
   {
     ServerImpl si = new ServerImpl();
     ClientImpl ci = new ClientImpl();
 
     new Thread(el).start();
 
-    el.start("localhost", port, si);
-    el.connect(new InetSocketAddress("localhost", port), ci);
+    el.start("localhost", 0, si);
+
+    SocketAddress address;
+    synchronized (si) {
+      if ((address = si.getServerAddress()) == null) {
+        si.wait();
+      }
+
+      address = si.getServerAddress();
+    }
+
+    logger.debug("Bound to address {}", address);
```

Comment:
Should the logging be done in the ServerTest.ServerImpl.registered() if at all? If logging is necessary it will benefit all test cases that use ServerImpl.